filters/include/pcl/filters/impl/voxel_grid_covariance.hpp

```diff
@@ -401,7 +398,38 @@ pcl::VoxelGridCovariance<PointT>::getNeighborhoodAtPoint (const PointT& referenc
     }
   }
 
-  return (static_cast<int> (neighbors.size ()));
+  return static_cast<int> (neighbors.size());
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////
+template<typename PointT> int
+pcl::VoxelGridCovariance<PointT>::getNeighborhoodAtPoint (const PointT& reference_point, std::vector<LeafConstPtr> &neighbors) const
+{
+  Eigen::MatrixXi relative_coordinates = pcl::getAllNeighborCellIndices();
+  return getNeighborhoodAtPoint(relative_coordinates, reference_point, neighbors);
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////
+template<typename PointT> int
+pcl::VoxelGridCovariance<PointT>::getVoxelAtPoint(const PointT& reference_point, std::vector<LeafConstPtr> &neighbors) const
+{
+  return getNeighborhoodAtPoint(Eigen::MatrixXi::Zero(3,1), reference_point, neighbors);
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////
+template<typename PointT> int
+pcl::VoxelGridCovariance<PointT>::getAdjacentVoxelsAtPoint(const PointT& reference_point, std::vector<LeafConstPtr> &neighbors) const
+{
+  Eigen::MatrixXi relative_coordinates(3, 7);
```

Comment:
Replace this with `Eigen::Matrix<int, 3, Eigen::Dynamic>`  to avoid building another temporary object (`Eigen::MatrixXi` -> `const Eigen::Matrix<int, 3, Eigen::Dynamic>&`) once you pass it as an argument to the function.

Reply:
```MatrixXi``` is replaced with ```Matrix<int, 3, Dynamic>```